### PR TITLE
Add test cases for TaskCommandParser in TaskCommandParserTest

### LINKED ISSUE #96: Add test cases for TaskCommandParser (TaskCommandParserTest)

### PR DESCRIPTION
**Pull request by wilsonsfh**
Fixes #96 

**Update**
General:
- Added logger to some classes
- Added defensive coding
- Improved general logic for TaskCommand task/, making it optional to add some task/ parameters 

1. Created test cases for TaskCommandParserTest.java
2. Messages.java - changed custom message for invalid task status
3. 
 - TaskCommand.java - Updated message usage when user inputs "task", which is an invalid command that prompts the comprehensive help message for TaskCommand
 - Added override for equals to check test cases 
4. ParserUtil.java - Defensive coding, to check edge case for task/ with empty task description
5. TaskCommandParser.java - Used optional, to ensure task/ prefix is used, otherwise return appropriate error to indicate that task/ is needed
6. Task.java - Added override for equals to check test cases
7. Taskstatus.java - Added method to convert string input of task status to the enum type of Task Status, to improve user input handling in ParserUtil.java (not applied in this PR)
8. ParserUtilTest.java - Added one test case to check parseTask() method works expectedly with valid value passed

Local error but pass GitHuB CI's check
Image

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/TaskCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        LocalDateTime expectedDate = LocalDateTime.of(2025, 12, 31, 23, 59);
+        Task expectedTask = new Task("", TaskStatus.YET_TO_START, expectedDate);
+        TaskCommand expectedCommand = new TaskCommand(Index.fromOneBased(1), expectedTask);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_validArgsWithDate_success() {
+        String userInput = "1 t/Finish assignment d/2025-12-31 23:59";
```
Comment (Jeremiah33-3) [inline src/test/java/seedu/address/logic/parser/TaskCommandParserTest.java:33]: our prefix for duedate is due/, so maybe that is why this test case might fail.

File: src/test/java/seedu/address/logic/parser/TaskCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        String userInput = "2 t/Buy groceries";
+        Task expectedTask = new Task("Buy groceries", TaskStatus.YET_TO_START, null);
+        TaskCommand expectedCommand = new TaskCommand(Index.fromOneBased(2), expectedTask);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_missingIndex_failure() {
+        String userInput = "t/Do homework d/2025-12-31 23:59";
```
Comment (Jeremiah33-3) [inline src/test/java/seedu/address/logic/parser/TaskCommandParserTest.java:52]: The command word here, should it just be "task"?

**Review by Jeremiah33-3 — COMMENTED** [review]
Overall, the code that failed CI checks seem to be caused by ParseException, so can check if the prefixes you are using are correct. Thank you!

**Comment by awc1116** [comment]
did not pass all the test cases in the test file

TaskCommandParserTest > parse_validArgsWithoutDate_success() FAILED
    java.lang.IllegalArgumentException at TaskCommandParserTest.java:47
        Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:47
            Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:47

TaskCommandParserTest > parse_invalidDateFormat_failure() FAILED
    org.opentest4j.AssertionFailedError at TaskCommandParserTest.java:60

TaskCommandParserTest > parse_noTaskDescription_success() FAILED
    java.lang.IllegalArgumentException at TaskCommandParserTest.java:28
        Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:28
            Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:28

TaskCommandParserTest > parse_validArgsWithDate_success() FAILED
    java.lang.IllegalArgumentException at TaskCommandParserTest.java:[38](https://github.com/AY2425S2-CS2103-F09-4/tp/actions/runs/14111359401/job/39530633243?pr=97#step:8:39)
        Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:38
            Caused by: seedu.address.logic.parser.exceptions.ParseException at TaskCommandParserTest.java:38

File: src/main/java/seedu/address/logic/parser/TaskCommandParser.java [hunk 3]
```diff
@@ -46,8 +46,7 @@ public TaskCommand parse(String args) throws ParseException {
                 throw new ParseException(MESSAGE_INCORRECT_DATE_FORMAT);
             }
         }
-
-        String taskDescription = argMultimap.getValue(PREFIX_TASK_DESC).get();
```
Comment (bkkuan) [inline src/main/java/seedu/address/logic/parser/TaskCommandParser.java:50]: Should this be extracted out?

**Comment by wilsonsfh** [comment]
I will look into the failed test cases. The PR got updated due to updates from my local repo. Thanks for the comments!

**Comment by wilsonsfh** [comment]
I have updated my original PR comment to reflect the changes I have done for this PR round. Do review and close it!

File: src/main/java/seedu/address/logic/Messages.java [hunk 4]
```diff
@@ -24,7 +24,8 @@ public class Messages {
             + "Use \"listtasks PERSON IDEX\" to check tasks index.";
     public static final String MESSAGE_EMPTY_TASK_DESC = "Task description cannot be empty! \n";
     public static final String MESSAGE_NO_TASK_FOR_MEM = "No tasks found for %s.";
-    public static final String MESSAGE_INCORRECT_TASK_STATUS = "No available task status.";
+    public static final String MESSAGE_INCORRECT_TASK_STATUS = "Task status can be only either: "
+                                                             + "yet to start | in progress | completed";
```
Comment (bkkuan) [inline src/main/java/seedu/address/logic/Messages.java:28]: I like the detailed information of this change

File: src/main/java/seedu/address/logic/commands/TaskCommand.java [hunk 5]
```diff
@@ -20,12 +18,19 @@
...
-            + PREFIX_TASK_DESC + "Complete project milestone "
-            + PREFIX_DUE_DATE + "2025-05-05 12:00";
+        + "Usage:\n"
+        + "  " + COMMAND_WORD + " INDEX task/TASK_DESCRIPTION[, DUE_DATE][, STATUS]\n"
+        + "\n"
+        + "Parameters:\n"
+        + "  INDEX            - A positive integer indicating the person's order in the list.\n"
+        + "  task/            - Task description (required).\n"
+        + "  DUE_DATE         - (Optional) In the format yyyy-MM-dd HH:mm\n"
+        + "  STATUS           - (Optional) One of: 'yet to start', 'in progress', 'completed'\n"
```
Comment (bkkuan) [inline src/main/java/seedu/address/logic/commands/TaskCommand.java:28]: I like how detailed the separation of the task, due date and task status is in the parameters

**Review by bkkuan — COMMENTED** [review]
LGTM
